### PR TITLE
feat: allow use of existing network resources

### PR DESCRIPTION
**Pull request by alannix-lw**
## Summary

This PR allows the user to specify existing network resources - subnet ID and security group ID - for use when deploying the AWLS scanning resources.

## How did you test this change?

Added example and validated with `make ci`

## Issue

[RAIN-48892](https://lacework.atlassian.net/browse/RAIN-48892)

[RAIN-48892]: https://lacework.atlassian.net/browse/RAIN-48892?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: variables.tf [hunk 1]
```diff
@@ -149,6 +149,30 @@ variable "use_existing_vpc" {
...
+variable "use_existing_security_group" {
+  type        = bool
+  default     = false
+  description = "Set this to `true` to use an existing security group for scanning compute resources."
+}
+
+variable "security_group_id" {
+  type        = string
+  default     = ""
+  description = "The ID of the security group to use for scanning compute resources.  Must also set `use_existing_security_group` to `true`."
```
Comment (theopolis) [inline variables.tf:161]: In the cases where we have two variables entangled then we can write a validation condition like this: https://github.com/lacework/terraform-aws-agentless-scanning/pull/57/files#diff-dc46acf24afd63ef8c556b77c126ccc6e578bc87e3aa09a931f33d9bf2532fbbR976

```
resource "null_resource" "check_security_group_id" {
  count = length(var.security_group_id) > 0 ? (var.use_existing_security_group ? 0 : "Error: When var.security_group_id is used then var.use_existing_security_group = true") : 0
}
```

File: variables.tf [hunk 2]
```diff
@@ -149,6 +149,30 @@ variable "use_existing_vpc" {
...
+variable "use_existing_subnet" {
+  type        = bool
+  default     = false
+  description = "Set this to `true` to use an existing subnet for scanning compute resources."
+}
+
+variable "subnet_id" {
+  type        = string
+  default     = ""
+  description = "The ID of the subnet to use for scanning compute resources.  Must also set `use_existing_subnet` to `true`."
```
Comment (theopolis) [inline variables.tf:173]: ditto to the above

**Review by theopolis — COMMENTED** [review]
Since we are gaining a lot of usages of the `use_existing` pattern I am worried about people setting some variables and not the corresponding `use_existing_*` and then encountering unexpected/unintended behavior.

Do you mind adding validation checks like the one I recommended in these cases?

File: variables.tf (hunk 1 above)
Comment (alannix-lw) [reply variables.tf:161]: Attempted to add the logic for this, but unless you have a static value for `security_group_id` - not a reference to a resource property - you'll get something like the following when running a plan:
```
│ Error: Invalid count argument
│ 
│   on ../../main.tf line 981, in resource "null_resource" "check_security_group_id":
│  981:   count = length(var.security_group_id) > 0 ? (var.use_existing_security_group ? 0 : "Error: When var.security_group_id is used then var.use_existing_security_group must also = true") : 0
│ 
│ The "count" value depends on resource attributes that cannot be determined until apply, so Terraform cannot predict how many instances will be created. To work around this, use the -target argument to first apply only the resources that the count depends on.
╵
```
In the associated example for this change, the necessary resources are created, and then referenced in the module configuration [here](https://github.com/lacework/terraform-aws-agentless-scanning/blob/594abb5b59ecab90bbdf5da223a2221e89eb3b6b/examples/single-account-existing-vpc-networking/main.tf#L71).  This causes the plan to fail unless the resources are known.  Given the spirit of this addition - that the resources should all exist - are we ok with that?  If so, I can just update the example.

Comment (theopolis) [reply variables.tf:161]: I see, dang, so there is no way to check if a variable is used and disregard the value (whether it is computed or not).

Sorry for the delay, let's go with your original proposal and drop the `null_resource`s

Comment (alannix-lw) [reply variables.tf:161]: Reverted change - can I get a

**Review by theopolis — APPROVED** [review]

**Pull request merged**